tests/tt_metal/tt_metal/dispatch/dispatch_program/test_ProgramReuse.cpp

```diff
@@ -0,0 +1,368 @@
+            vector<uint32_t> dummy_kernel0_args_readback;
+            tt::tt_metal::detail::ReadFromDeviceL1(
+                device,
+                core_coord,
+                rta_base_dm0,
+                num_runtime_args_for_cr1 * sizeof(uint32_t),
+                dummy_kernel0_args_readback);
+            pass &= (dummy_cr1_args == dummy_kernel0_args_readback);
+
+            vector<uint32_t> dummy_kernel1_args_readback;
+            tt::tt_metal::detail::ReadFromDeviceL1(
+                device,
+                core_coord,
+                rta_base_dm1,
+                num_runtime_args_for_cr1 * sizeof(uint32_t),
+                dummy_kernel1_args_readback);
+            pass &= (dummy_cr1_args == dummy_kernel1_args_readback);
+
+            vector<uint32_t> dummy_compute_args_readback;
+            tt::tt_metal::detail::ReadFromDeviceL1(
+                device,
+                core_coord,
+                rta_base_compute,
+                num_runtime_args_for_cr1 * sizeof(uint32_t),
+                dummy_compute_args_readback);
+            pass &= (dummy_cr1_args == dummy_compute_args_readback);
+        }
+        vector<uint32_t> semaphore_vals;
+        for (const CoreCoord& core_coord : sem_cr_range) {
+            uint32_t expected_sem_idx = 0;
+            uint32_t sem_base_addr = program->get_sem_base_addr(devices_[0], core_coord, CoreType::WORKER);
+            tt::tt_metal::detail::ReadFromDeviceL1(
+                device, core_coord, sem_base_addr, semaphore_buffer_size, semaphore_vals);
+            for (uint32_t i = 0; i < semaphore_vals.size();
+                 i += (hal.get_alignment(HalMemType::L1) / sizeof(uint32_t))) {
+                pass &= (semaphore_vals[i] == dummy_sems[expected_sem_idx]);
+                expected_sem_idx++;
+            }
+        }
+        vector<uint32_t> cb_config_vector;
+        for (const CoreCoord& core_coord : sem_cr_range) {
+            tt::tt_metal::detail::ReadFromDeviceL1(
+                device,
+                core_coord,
+                program->get_cb_base_addr(device, core_coord, CoreType::WORKER),
+                cb_config_buffer_size,
+                cb_config_vector);
+            uint32_t cb_addr = device->get_base_allocator_addr(HalMemType::L1);
+            for (uint32_t i = 0; i < cb_config.cb_config_vector.size(); i++) {
+                const uint32_t index = cb_config.cb_config_vector[i].cb_id * sizeof(uint32_t);
+                const uint32_t cb_num_pages = cb_config.cb_config_vector[i].num_pages;
+                const uint32_t cb_size = cb_num_pages * cb_config.cb_config_vector[i].page_size;
+                const bool addr_match = cb_config_vector.at(index) == cb_addr;
+                const bool size_match = cb_config_vector.at(index + 1) == cb_size;
+                const bool num_pages_match = cb_config_vector.at(index + 2) == cb_num_pages;
+                pass &= (addr_match and size_match and num_pages_match);
```

Comment:
Use `EXPECT_EQ(b_config_vector.at(index), cb_addr)` on every assertion? Test failure will show which line failed, instead of getting to the end with generic `pass != true`

Reply:
Done